src/backend/shortcuts/nonesteamgame/nonesteamgame.ts

```diff
@@ -173,17 +179,36 @@ function checkIfAlreadyAdded(object: Partial<ShortcutObject>, title: string) {
 
 /**
  * Adds a non-steam game to steam via editing shortcuts.vdf
- * @param steamUserdataDir Path to steam userdata directory
  * @param gameInfo @see GameInfo of the game to add
+ * @param steamUserdataDir Path to steam userdata directory, optional
+ * @param bkgDataUrl data url string for the background image, optional
+ * @param bigPicDataUrl data url of the Big Picture image, optional
  * @returns boolean
  */
 async function addNonSteamGame(props: {
-  steamUserdataDir: string
   gameInfo: GameInfo
-  bkgDataUrl: string
-  bigPicDataUrl: string
+  steamUserdataDir?: string
+  bkgDataUrl?: string
+  bigPicDataUrl?: string
 }): Promise<boolean> {
-  const { folders, error } = checkSteamUserDataDir(props.steamUserdataDir)
+  const steamUserdataDir =
+    props.steamUserdataDir || (await getSteamUserdataDir())
+
+  let bkgDataUrl = props.bkgDataUrl
+  let bigPicDataUrl = props.bigPicDataUrl
+
+  if (bkgDataUrl === undefined || bigPicDataUrl === undefined) {
+    const window = BrowserWindow.getAllWindows()[0]
+
+    bkgDataUrl = await window.webContents.executeJavaScript(
+      `window.imageData("${props.gameInfo.art_cover}", 1920, 620)`
```

Comment:
I made the `imageData` method global in the window object so we can ask the frontend to generate the image on demand instead of passing them with the click event. This is needed since the after install has no context of a user click to receive these images.